### PR TITLE
#58/swteam/openid registration

### PR DESCRIPTION
**Pull request by fpottbaecker**
This will add the ability to sign in with an unlinked openID account, which prompts the user to fill in missing model data and creates an account once he submits the form.

This pull request also includes an initial german translation for the devise language file, as requested by the PO.

### CODE REVIEW

**Comment by fpottbaecker** [comment]
This feature has been reviewed and accepted by the product owner, @MerlindlH.

**Review by DerIntergalaktischeGrosshorst — APPROVED** [review]
Looks fine

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -29,6 +28,17 @@ def sign_in_user(user)
       flash_if_navigational :notice, :success, provider: 'OpenID'
     end
 
+    def sign_up_user(auth)
+      session['omniauth.data'] = {
+        uid: auth.uid,
+        provider: auth.provider,
+        email: auth.info.email,
+        expires: Time.current + 10.minutes
```
Comment (ErikBrendel) [inline app/controllers/users/omniauth_callbacks_controller.rb]: why 10 minutes?

File: app/models/user.rb [hunk 2]
```diff
@@ -47,7 +47,28 @@ def reset_omniauth
     self.provider = nil
   end
 
-  def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first
+  class << self
+    def new_with_session(_, session)
```
Comment (ErikBrendel) [inline app/models/user.rb:51]: please add a comment: What is a session in this context

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -21,24 +21,26 @@
...
-  <div class="form-group">
-    <%= f.label :password_confirmation, :class => 'control-label col-lg-2' %>
-    <div class="col-lg-10">
-      <%= f.password_field :password_confirmation, class: 'form-control' %>
-      <%= f.error_span(:password_confirmation) %>
-    </div>
-  </div>
+  <% unless resource.has_omniauth? %>
+      <div class="form-group">
+        <%= f.label :password, :class => 'control-label col-lg-2' %>
```
Comment (ErikBrendel) [inline app/views/devise/registrations/new.html.erb]: :rocket: (also in line +7)

File: features/openid/registration.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
+Feature: OpenID Registration
+  In order to simplify my login process
+  As an unregistered user with an OpenID Account
+  I want to sign up with my OpenID Account
+
+  Scenario: Sign up attempt
+    Given a new user
+    And an unlinked OpenID account with email 'test@test.com'
+    When the account is used to sign in
+    Then the 'E-Mail' input should be filled with 'test@test.com'
```
Comment (ErikBrendel) [inline features/openid/registration.feature]: `should already be filled with` maybe? Should be filled sounds like an action (like 'the user should fill in now')

**Review by ErikBrendel — APPROVED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (fpottbaecker) [reply app/models/user.rb:51]: This method is called by devise when a resource is built, which is done automatically by most controllers, or in this case the users_controller, with the call to `load_resource` call

**Pull request merged**